### ISSUE TITLE
Spec compliance: Multiple resources should not be accepted on POST or PATCH

### ISSUE THREAD

**Issue by carlbennettnz** [ethanresnick/json-api#65]
On `POST`

> A resource can be created by sending a POST request to a URL that represents a collection of resources. The request MUST include a single resource object as primary data.

On `PATCH`

> The PATCH request MUST include a single resource object as primary data.

This feature should only be available as part of the [bulk extension](http://jsonapi.org/extensions/bulk/), and in that case:

> […] any request that involves multiple operations MUST only succeed if all operations are performed successfully. The state of the server MUST NOT be changed by a request if any individual operation fails.

**Comment by carlbennettnz** [comment]
Same goes for `DELETE` requests.

**Comment by ethanresnick** [comment]
You're totally right about this, but it's requires a lot of code to fix (e.g. to simulate transactions in Mongo) with little immediate benefit, so I probably won't address it for a while. Also, the JSON API extension system is likely to [change radically](https://github.com/json-api/json-api/pull/650) in 1.1, so I think I want to let that settle before refactoring for this.

**Comment by carlbennettnz** [comment]
Sounds good to me :+1:

**Comment by slothbear** [comment]
I'm looking for info (in general) regarding updating multiple records via PATCH. Could you link to the spec compliance document you mentioned?
